tests/python/relay/test_op_level1.py

```diff
@@ -217,6 +217,18 @@ def test_concatenate():
     zz = run_infer_type(z)
     assert zz.checked_type == relay.TensorType((n, t + t, 100))
 
+    # check shape mismatches (the following case is expected to raise tvm._ffi.base.TVMError.
+    try:
+        x = relay.var('p1', shape=(2, 5))
+        y = relay.var('p2', shape=(2, 3))
+        c = relay.concatenate([x, y], axis=0)
+        func = relay.Function([x, y], c)
+        relay_module = relay.Module.from_expr(func)
```

Comment:
Are you implicitly inferring type here? Should we have some assertions?

Reply:
Can you explain more here? Current action here is similar to `run_infer_type()` used in other tests.

Reply:
Yeah, I am saying that type inference implicit here when creating a module. We probably prefer to invoke `run_infer_type` directly and check the expected type.

Reply:
Fixed. But one weird thing I met is, if I use `run_infer_type()`, then no error is thrown

```python
# no error 
x = relay.var('p1', shape=(2, 5))
y = relay.var('p2', shape=(2, 3))
z = relay.concatenate([x, y], axis=1)
zz = run_infer_type(z)

# throws tvm.Error
x = relay.var('p1', shape=(2, 5))
y = relay.var('p2', shape=(2, 3))
z = relay.concatenate([x, y], axis=1)
func = relay.Function([x, y], c)
zz = run_infer_type(func)
```

While `run_infer_type()` is 

```python
def run_infer_type(expr):
    mod = relay.Module.from_expr(expr)
    mod = transform.InferType()(mod)
    entry = mod["main"]
    return entry if isinstance(expr, relay.Function) else entry.body
```

It seems we have to pack it into a `relay.Function` then the error can be detected.

Reply:
That seems very strange, since `module.from_expr` is supposed to pack a non-function into a function (see https://github.com/dmlc/tvm/blob/master/src/relay/ir/module.cc#L232).

Reply:
I checked out your branch and tried running the example you gave myself and did not get an error in either case so I'm not sure what you're referring to. Was there a problem before that you fixed?